app/views/devise/two_factor_authentication/show.html.erb

```diff
@@ -7,5 +11,9 @@
   <%= submit_tag "Submit" %>
 <% end %>
 
-<%= link_to 'Resend Code', resend_code_user_two_factor_authentication_path, action: :get %>
-<%= link_to "Sign out", destroy_user_session_path, :method => :delete %>
\ No newline at end of file
+<% if current_user.direct_otp %>
+<%= link_to "Resend Code", resend_code_user_two_factor_authentication_path, action: :get %>
+<% else %>
+<%= link_to "Send me a code", resend_code_user_two_factor_authentication_path, action: :get %>
+<% end %>
```

Comment:
The idea is that TOTP users can always choose to have a code sent to them if they don't have their app.  (all of the TOTP sites I've tried always let you fall back to SMS, for better or for worse).